src/PAModel/Serializers/TransformResourceJson.cs

```diff
@@ -0,0 +1,141 @@
+            {
+                var fileName = file.Key.GetFileName();
+                if (!app._resourcesJson.Resources.Any(x => x.FileName == fileName) && logoFileName != fileName)
+                {
+                    localFileResourceJsonEntries.Add(GenerateResourceJsonEntryFromAssetFile(file.Key));
+                }
+            }
+            app._resourcesJson.Resources = app._resourcesJson.Resources.Concat(localFileResourceJsonEntries).ToArray();
+
+            // bring the order of resourceJson back to avoid checksum violation.
+            if (app._entropy?.ResourceJsonIndices != null && app._entropy.ResourceJsonIndices.Count > 0)
+            {
+                var orderedResourcesList = new List<ResourceJson>();
+                var orderedIndices = app._entropy.ResourceJsonIndices.OrderBy(x => x.Value);
+                foreach (var kvp in orderedIndices)
+                {
+                    var resource = app._resourcesJson.Resources.Where(x => x.Name == kvp.Key);
+                    orderedResourcesList.Add(resource.SingleOrDefault());
+                }
+
+                // handle the cases when some new files were added to the asset folder offline. The entries for the new assets would go at the end, after all the ordered resources have been added.
+                orderedResourcesList.AddRange(app._resourcesJson.Resources.Where(x => !app._entropy.ResourceJsonIndices.ContainsKey(x.Name)));
+                app._resourcesJson.Resources = orderedResourcesList.ToArray();
+            }
+        }
+
+        // create the resource json entry dynamically from the asset file.
+        private static ResourceJson GenerateResourceJsonEntryFromAssetFile(FilePath filePath)
+        {
+            var fileName = filePath.GetFileName();
+            var contentKind = GetContentKind(fileName);
+            return new ResourceJson()
+            {
+                Name = filePath.GetFileNameWithoutExtension(),
+                Content = GetContentKind(fileName),
+                Schema = GetSchema(contentKind),
+                ResourceKind = ResourceKind.LocalFile,
+                Path = FilePath.RootedAt("Assets", filePath).ToMsAppPath(),
+                FileName = fileName,
+                Type = "ResourceInfo",
+                IsSampleData = false,
+                IsWritable = false
+            };
+        }
+
+        private static ContentKind GetContentKind(string name)
+        {
+            if (ImageExtensionRegEx.IsMatch(name))
+                return ContentKind.Image;
+            else if (AudioExtensionRegEx.IsMatch(name))
+                return ContentKind.Audio;
+            else if (VideoExtensionRegEx.IsMatch(name))
+                return ContentKind.Video;
+            else if (PdfExtensionRegEx.IsMatch(name))
+                return ContentKind.Pdf;
+            else
+                return ContentKind.Unknown;
+        }
+
+        private static string GetSchema(ContentKind contentKind)
```

Comment:
If you update the schema to use the enum, this gets simpler.

Reply:
the problem is I can't have '?' in an enum, which is the value of schema used by server for unknown types, now sure if we can ever have such type...